CoreScriptsRoot/Modules/Server/ServerChat/ChatChannel.lua

```diff
@@ -197,6 +234,22 @@ function methods:GetHistoryLog()
 	return DeepCopy(self.ChatHistory)
 end
 
+function methods:GetHistoryLogForSpeaker(speaker)
+	local userId = -1
+	local player = speaker:GetPlayer()
+	if player then
+		userId = Player.UserId
+	end
+	local chatlog = {}
+	for i = 1, #self.ChatHistory do
+		local logUserId = self.ChatHistory[i].SpeakerUserId
+		if self:CanCommunicateByUserId(userId, logUserId) then
```

Comment:
Won't this make a lot of requests, or does it automatically cache the value for all players that have joined that server? I also thought this only worked for in-game players.